jeopardy.js

```diff
@@ -0,0 +1,248 @@
+    return catIdArr;
+}
+
+
+
+
+
+/** Return object with data about a category:
+ *
+ *  Returns { title: "Math", clues: clue-array }
+ *
+ * Where clue-array is:
+ *   [
+ *      {question: "Hamlet Author", answer: "Shakespeare", showing: null},
+ *      {question: "Bell Jar Author", answer: "Plath", showing: null},
+ *      ...
+ *   ]
+ */
+
+async function getCategory(catId) {
+    let cat = await axios.get(`http://jservice.io/api/category?id=${catId}`)
+
+    return cat.data;
+
+}
+
+
+
+// -- Dom Logic ---
+
+function createTable() {
+    const theadLength = 6;
+    const width = 6;
+    const table = document.createElement('table');
+    table.setAttribute('id', 'jeopardy');
+    const thead = document.createElement('thead');
+    const trHead = document.createElement('tr');
+    const trBody = document.createElement('tr')
+    const tBody = document.createElement('tbody')
+
+
+    for (let i = 0; i < theadLength; i++) {
+        const th = document.createElement('th');
+        th.innerHTML = "people"
+        trHead.append(th);
+
+
+    }
+    thead.append(trHead);
+
+    table.append(thead);
+
+
+
+    for (let i = 0; i < theadLength; i++) {
+        const td = document.querySelector('td');
+        trBody.append(td);
+        // for (let i = 0; i < width; i ++){
+
+        //     }
+
+    }
+
+
+
+
+    //     for (let i = 0; i < width; i++) {
+    //         const header = document.createElement('th');
+    //         header.setAttribute('id', 'catNames');
+    //         // header.innerHTML = categoriesNew;
+    //         thead.append(header);
+
+
+
+    //     }
+    //     table.append(thead);
+
+    //     for (let j = 0; j < height - 1; j++) {
+    //         const row = document.createElement('tr')
+    //         row.innerHTML = "?";
+    //         for (let y = 0; y < width - 1; y++) {
+    //             const cell = document.createElement('td')
+    //             cell.innerHTML = "?";
+    //             row.append(cell);
+    //         }
+    //         table.append(row);
+    //         mainTable.append(table);
+    //     }
+
+    // }
+
+
+
+}
+function getTable() {
+    const checkTable = document.querySelector('table')
+    if (!checkTable) {
+        return createTable();
+    }
+    return table;
+
+}
+
+/** Fill the HTML table#jeopardy with the categories & cells for questions.
+ *
+ * - The <thead> should be filled w/a <tr>, and a <td> for each category
+ * - The <tbody> should be filled w/NUM_QUESTIONS_PER_CAT <tr>s,
+ *   each with a question for each category in a <td>
+ *   (initally, just show a "?" where the question/answer would go.)
+ */
+
+function fillTable() {
+    const trHead = document.createElement('tr');
+    const table = document.querySelector('#gameBoard')
+    // query selector to find the header with id 
+    // loop through headers 
+
+    for (cat of categories) {
+        console.log(cat);
+        for (let i = 0; i < 6; i++) {
+            const th = document.createElement('th');
```

Comment:
I suggest doing the creation of the table in the first loop (i.e. line 174) and perhaps as a separate method

```
function createHeader (title) {
   const header = document.createElement('th')
   header.setAttribute('id', "title-" + title)
   header.innerHTML = title
}
```